flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java

```diff
@@ -635,8 +644,18 @@
 	 * The default number of archived jobs for the jobmanager
 	 */
 	public static final int DEFAULT_JOB_MANAGER_WEB_ARCHIVE_COUNT = 5;
-	
-	
+
+	/**
+	 * By default, submitting jobs from the web-frontend is allowed.
+	 */
+	public static final boolean DEFAULT_JOB_MANAGER_WEB_SUBMISSION = true;
+
+	/**
+	 * Default directory for uploaded file storage for the Web frontend.
+	 */
+	public static final String DEFAULT_JOB_MANAGER_WEB_UPLOAD_DIR =
+			(System.getProperty("java.io.tmpdir") == null ? "/tmp" : System.getProperty("java.io.tmpdir")) + "/webmonitor/";
```

Comment:
How about handling the temp directory similar to the IOManager and BlobManager: We create directory in the temporary directory (with a random name) and use a shutdown hook to remove the directory on shutdown again.

I don't like keeping the jar files because it will pollute the directory on the server.
Also the current approach does not allow running multiple JobManagers at the same machine (something that can easily happen with YARN).